smart_open/s3.py

```diff
@@ -114,6 +114,16 @@ def seek(self, position):
         self._position = position
         range_string = _range_string(self._position)
         logger.debug('content_length: %r range_string: %r', self._content_length, range_string)
+
+        #
+        # Close old body explicitly.
+        # When first seek(), self._body is not exist. Catch the exception and do nothing.
+        #
+        try:
+            self._body.close()
+        except Exception as e:
```

Comment:
Why not be explicit and catch AttributeError?

Reply:
I'm +1 too for catching more "restricted", after this fix I think we can merge.

Reply:
Is really only `AttributeError`? Can `close()` raise something else?

Reply:
@piskvorky I don't think we expect close itself to raise anything.

I think the intention here is to mimic the behavior of:

```python
if body is not None:
    body.close()
```

without paying the price of the conditional every time.  I think the author is assuming that most the time, the body will be non-null. I think this assumption is correct.